servers/src/main/java/tachyon/master/file/meta/TTLBucketList.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master.file.meta;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+import tachyon.Constants;
+
+/**
+ * A list of non-empty {@link TTLBucket}s sorted by ttl interval start time of each bucket.
+ * <p/>
+ * Two adjacent buckets may not have adjacent intervals since there may be no files with ttl value
+ * in the skipped intervals.
+ * <p/>
+ * Thread-safety is guaranteed by {@link ConcurrentSkipListSet}.
+ */
+public final class TTLBucketList {
+  /**
+   * List of buckets sorted by interval start time. SkipList is used for O(logn) insertion and
+   * retrieval, see {@link ConcurrentSkipListSet}.
+   */
+  private final ConcurrentSkipListSet<TTLBucket> mBucketList;
+
+  public TTLBucketList() {
+    mBucketList = new ConcurrentSkipListSet<TTLBucket>();
+  }
+
+  /**
+   * Inserts an {@link InodeFile} to the appropriate bucket where its ttl end time lies in the
+   * bucket's interval, if no appropriate bucket exists, a new bucket will be created to contain
+   * this file, if ttl value is {@link Constants#NO_TTL}, the file won't be inserted to any buckets
+   * and nothing will happen.
+   *
+   * @param file the file to be inserted
+   */
+  public void insert(InodeFile file) {
+    if (file.getTTL() == Constants.NO_TTL) {
+      return;
+    }
+
+    long ttlEndTimeMs = file.getCreationTimeMs() + file.getTTL();
+    // Gets the last bucket with interval start time less than or equal to the file's life end
+    // time.
+    TTLBucket bucket = mBucketList.floor(new TTLBucket(ttlEndTimeMs));
```

Comment:
I am little bit concerned about creating a new bucket each time for lookup an existing bucket.
Maybe the comparator approach is more suitable.

But, I am fine with leaving a TODO here to address this later.